topology_test.go

```diff
@@ -0,0 +1,158 @@
+package duty
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/flanksource/duty/fixtures/dummy"
+	"github.com/flanksource/duty/models"
+	_ "github.com/flanksource/duty/types"
+	ginkgo "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+// For debugging
+// nolint
+func prettytree(mytree []*models.Component) {
+	for _, c := range mytree {
+		fmt.Printf("- %s {analysis: %v}\n\n", c.Name, c.Summary)
+		for _, cc := range c.Components {
+			fmt.Printf("  |- %s {analysis: %v}\n\n", cc.Name, cc.Summary)
+			for _, ccc := range cc.Components {
+				fmt.Printf("    |- %s {analysis: %v}\n\n", ccc.Name, ccc.Summary)
+				for _, cccc := range ccc.Components {
+					fmt.Printf("      |- %s {analysis: %v}\n\n", cccc.Name, cccc.Summary)
+				}
+			}
+		}
+	}
+}
+
+var _ = ginkgo.Describe("Topology behavior", ginkgo.Ordered, func() {
+
+	ginkgo.It("Should create root tree", func() {
+		tree, err := QueryTopology(TopologyOptions{})
+		Expect(err).ToNot(HaveOccurred())
+
+		treeJSON, err := json.Marshal(tree)
+		Expect(err).ToNot(HaveOccurred())
+
+		expected := readTestFile("fixtures/expectations/topology_root_tree.json")
+		Expect(expected).Should(MatchJSON(string(treeJSON)))
```

Comment:
this could probably refactored into `test(options, file)`

Reply:
👍🏽